File: src/parsing/expressions/exists_subquery.ts
import SubqueryExpression from "./subquery_expression";

/** `EXISTS { ... }` — true when the subquery produces at least one row. */
class ExistsSubquery extends SubqueryExpression {
    protected reduce(_rows: any[], count: number): boolean {
        return count > 0;
    }
}

export default ExistsSubquery;

File: flowquery-py/src/parsing/expressions/exists_subquery.py
"""EXISTS { ... } subquery expression."""

from typing import Any, List

from .subquery_expression import SubqueryExpression


class ExistsSubquery(SubqueryExpression):
    """`EXISTS { ... }` -- True when the subquery produces at least one row."""

    def reduce(self, rows: List[Any], count: int) -> Any:
        return count > 0
